src/core/server/saved_objects/routes/create.ts

```diff
@@ -44,15 +44,16 @@ export const registerCreateRoute = (router: IRouter) => {
               })
             )
           ),
+          initialNamespaces: schema.maybe(schema.arrayOf(schema.string(), { minSize: 1 })),
```

Comment:
Well, everywhere else that we use the `namespace` or `namespaces` field, we are specifying _what namespace(s) to search for an object in_.

It seems to me that we should make this distinct, since this is _what namespace(s) to create this object in_. I know this is only used for the "create" and "bulkCreate" methods, but still.

I lean towards leaving it as-is, maybe @pgayvallet has an opinion to share?

Reply:
Ah, I had [somehow] forgotten about the `bulk_update` changes we _just_ made where we use `namespace` to denote where to locate the object, as opposed to listing the namespaces the updated object should be included in.

I'll buy that argument, but also happy to defer to platform since they own the APIs

Reply:
I get the distinction between namespaces parameters that are used to locate the object(s) and parameters that are used to define the values to set, but as adding/removing namespaces is done with specific APIs. I doubt we will ever have two 'namespaces' parameters in a route / so API, so I would just use the `namespaces` name for create/bulkCreate.

But that's not a strong opinion. @rudolf, WDYT?

Reply:
I think the behaviour of setting something on create and finding inside a namespace on get is intuitive enough.

But it seems like we can delete an object from an existing namespace when you set `initialNamespaces: 'new space'` and `namespace: 'the space the object exists in'` with `overwrite=true`.

I find it difficult to think what the behaviour of initialNamespaces with overwrite=true should be. It feels like you should only be able to overwrite if you're not touching the namespaces (initialNamespaces = []). I can't remember, why did we want to restrict namespace operations to namespace-specific methods like deleteFromNamespace? It feels like similar reasoning should apply to bulkCreate (even with overwrite=true)

Reply:
> I think the behaviour of setting something on create and finding inside a namespace on get is intuitive enough.

Fair enough!

> But it seems like we can delete an object from an existing namespace when you set `initialNamespaces: 'new space'` and `namespace: 'the space the object exists in'` with `overwrite=true`.

Yes you can overwrite an object and change its namespace by using overwrite=true. This is by design.

> I find it difficult to think what the behaviour of initialNamespaces with overwrite=true should be. It feels like you should only be able to overwrite if you're not touching the namespaces (initialNamespaces = []).

Well, Elasticsearch treats a create+overwrite as if you're creating a new document and blowing the old one away, so my opinion is we shouldn't artificially limit the behavior here -- there are potentially valid use cases where we might want to recreate an existing object in "all spaces", for instance.

This could result in a scenario that's a bit counter-intuitive: If I am a user who can only access the "Foo" space, I can overwrite a saved object that exists in the "Foo" and "Bar" spaces using `initialNamespaces: ['foo']` (effectively unsharing it from the "Bar" space). But this outcome is no different than deleting and recreating the object, which the user is also allowed to do.

> I can't remember, why did we want to restrict namespace operations to namespace-specific methods like deleteFromNamespace?

Several reasons: 1. At the time, `namespace`/`namespaces` was an implementation detail that was completely hidden from consumers, 2. I didn't think it was a good idea to conflate the concept of a namespace with normal object attributes that can be changed via an "update" operation, 3. Having dedicated "addToNamespaces" and "deleteFromNamespaces" APIs for this makes authorization checks a lot simpler to implement and easier to test / reason about, and 4. We may eventually want to allow creation of roles that are not allowed to share/unshare objects. (I guess we'd have to revisit the create+overwrite situation if we wanted to actually implement this)